lasertag_ppo.py

```diff
@@ -0,0 +1,352 @@
+from copy import deepcopy
+from typing import TypeVar
+
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from gymnasium import spaces
+from torch.distributions.categorical import Categorical
+from tqdm.auto import tqdm
+
+from lasertag import LasertagCorridor1
+from syllabus.core import Curriculum, TaskWrapper
+from syllabus.task_space import TaskSpace
+
+ObsType = TypeVar("ObsType")
+ActionType = TypeVar("ActionType")
+AgentID = TypeVar("AgentID")
+
+
+class LasertagParallelWrapper(TaskWrapper):
+    """
+    Wrapper ensuring compatibility with the PettingZoo Parallel API.
+
+    Lasertag Environment:
+        * Action shape:  `n_agents` * `Discrete(5)`
+        * Observation shape: Dict('image': Box(0, 255, (`n_agents`, 3, 5, 5), uint8))
+    """
+
+    def __init__(self, n_agents, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.n_agents = n_agents
+        self.task = None
+        self.episode_return = 0
+        self.task_space = TaskSpace(spaces.MultiDiscrete(np.array([[2], [5]])))
+        self.possible_agents = np.arange(self.n_agents)
```

Comment:
Ideally this should be a list of agent names
```suggestion
        self.possible_agents = [f"agent_{i}" for i in range(self.n_agents)]
```